ground motion functions and python data files

eusefalee opened the pull request:
ref #414

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1144312/) on 823c1e8 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/mastodon/docs/PRs/418/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/mastodon/docs/PRs/418/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 74d630f6e5889e568ec2d57dfc012ac6b3349b49`

moosebuild commented:
Job [Test](https://civet.inl.gov/job/1144316/) on 5c315bb wanted to post the following:

View the site [here](https://mooseframework.inl.gov/mastodon/docs/PRs/418/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1145005/) on a4868f8 wanted to post the following:


## Coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>74d630</th>
      <th colspan=4>#418 a4868f</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>93.84%</td>
      <td>93.89%</td>
      <td>+0.05%</td>
      <td>97.87%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3442</td>
      <td>3488</td>
      <td>+46</td>
      <td>46</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>226</td>
      <td>227</td>
      <td>+1</td>
      <td>1</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/mastodon/docs/PRs/418/coverage_diff)

[Full coverage report](https://mooseframework.inl.gov/mastodon/docs/PRs/418/coverage)



This comment will be updated on new commits.

somu15 commented on doc/content/source/functions/GroundMotionSim.md:
```diff
@@ -0,0 +1,37 @@
+                      'yaxis':{'title':'Acceleration (m/s<sup>2</sup>)'},
+                      'title':'Simulated Ground Motion'}
+              caption=Time series of a sample ground motion
+              id=sample
```
Can you also show the frequency content of this GM?

somu15 commented on src/functions/GroundMotionSim.C:
```diff
@@ -0,0 +1,154 @@
+  std::string nstr = std::to_string(n);
+
+  // Get path to simulation script
+  std::string PATH = getDataFileNameByName("GMSim.py");
```
What if the user has GMSim.py at multiple locations? How does this code pick which one to use then?

eusefalee replied:
The user should not have it in multiple locations. As part of this merge, a new folder called "data" is added in the top level folder. This is where GMSim.py is contained. It is not a user created script, it is the script that generates the actual ground motion.

somu15 commented on data/GMSim.py:
```diff
@@ -0,0 +1,441 @@
+# import csv
+np.set_printoptions(threshold=sys.maxsize)
+
+
```
Can you please add comments to each function and other important stuff in this Python file? Would help future users.

somu15 replied:
Oh I guess you already did this, right?

eusefalee replied:
Yes, the GMSim.py script is pretty well-documented

somu15 commented:
@eusefalee This looks great! I made some comments. We can merge it once you address these.

somu15 commented:
@cbolisetti Can you please merge this?

cbolisetti commented on data/GMSim.py:
```diff
@@ -0,0 +1,434 @@
+import numpy as np
+from scipy.stats import gamma
+import sys
+# import csv
```
Can you please remove this line if not used?

cbolisetti commented on data/GMSim.py:
```diff
@@ -0,0 +1,434 @@
+from re import T
```
Can you put this file in the `python/mastodonutils` folder instead and create unit tests for it? We have an existing `mastodonutils` python module and this would fit there perfectly fine. Also, please look at the `python/mastodonutils/test` folder for unit test examples.

eusefalee replied:
The GMSim.py needs to be in mastodon/data because the getDataFileNameByName function cannot find it in the mastodonutils

cbolisetti commented on data/GMSim.py:
```diff
@@ -0,0 +1,434 @@
+np.set_printoptions(threshold=sys.maxsize)
+
+
+def RJB2REPI(M, R):
```
Can you please change these comments to into docstrings? This will help users query them later. Its just a simple change that goes like this (see help on docstrings [here](https://realpython.com/documenting-python-code/)): 

```
class SimpleClass:
    """Class docstrings go here."""

    def say_hello(self, name: str):
        """Class method docstrings go here."""

        print(f'Hello {name}')
```

eusefalee replied:
Just the function definition comments, right?

cbolisetti replied:
Yes, that's right.

cbolisetti reviewed (changes requested):
@eusefalee and @somu15 thanks for this important piece of work! This looks great and I just have some minor comments. Let me know if you have any questions.

somu15 commented:
@eusefalee Are you done addressing the comments?

eusefalee commented:
Yes

> On Aug 22, 2022, at 1:05 PM, Som Dhulipala ***@***.***> wrote:
> 
> ﻿
> @eusefalee Are you done addressing the comments?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.
